src/provision-dialog.ts

```diff
@@ -241,9 +244,23 @@ class ProvisionDialog extends LitElement {
   protected updated(changedProps: PropertyValues) {
     super.updated(changedProps);
 
+    if (
+      changedProps.has("_state") ||
+      (this._state !== "IMPROV-STATE" &&
+        changedProps.has("_improvCurrentState"))
+    ) {
+      const state =
+        this._state === "IMPROV-STATE" && this._improvCurrentState
+          ? (ImprovCurrentState[
+              this._improvCurrentState
+            ] as keyof typeof ImprovCurrentState)
+          : this._state;
```

Comment:
In this case the state could end up being `IMPROV-STATE` ? that wouldn't work I think

Reply:
Only if `this._state` can be `IMPROV-STATE` while `this._improvCurrentState` is undefined... I don't think that should be possible?

Reply:
But I can throw an `UNKNOWN` state in between?

Reply:
Maybe we shouldn't fire it because the improv state will come in very soon ? Or actually, judging from the code, can this even happen ?

Reply:
No, it should not be possible... `ImprovCurrentState` is set before `_state` is set to `IMPROV-STATE`